pkg/resources/stream_on_directory_table_acceptance_test.go

```diff
@@ -320,6 +321,46 @@ func TestAcc_StreamOnDirectoryTable_CopyGrants(t *testing.T) {
 	})
 }
 
+func TestAcc_StreamOnDirectoryTable_RecreateWhenStale(t *testing.T) {
+	_ = testenvs.GetOrSkipTest(t, testenvs.EnableAcceptance)
+	acc.TestAccPreCheck(t)
+	resourceName := "snowflake_stream_on_directory_table.test"
+
+	schema, cleanupSchema := acc.TestClient().Schema.CreateSchemaWithOpts(t,
+		acc.TestClient().Ids.RandomDatabaseObjectIdentifierInDatabase(acc.TestClient().Ids.DatabaseId()),
+		&sdk.CreateSchemaOptions{
+			DataRetentionTimeInDays:    sdk.Pointer(0),
+			MaxDataExtensionTimeInDays: sdk.Pointer(0),
+		},
+	)
+	t.Cleanup(cleanupSchema)
+	id := acc.TestClient().Ids.RandomSchemaObjectIdentifierInSchema(schema.ID())
+
+	stage, cleanupStage := acc.TestClient().Stage.CreateStageWithDirectory(t, acc.TestClient().Ids.SchemaId())
+	t.Cleanup(cleanupStage)
+
+	model := model.StreamOnDirectoryTable("test", id.DatabaseName(), id.Name(), id.SchemaName(), stage.ID().FullyQualifiedName())
+	resource.Test(t, resource.TestCase{
+		ProtoV6ProviderFactories: acc.TestAccProtoV6ProviderFactories,
+		TerraformVersionChecks: []tfversion.TerraformVersionCheck{
+			tfversion.RequireAbove(tfversion.Version1_5_0),
+		},
+		CheckDestroy: acc.CheckDestroy(t, resources.StreamOnDirectoryTable),
+		Steps: []resource.TestStep{
+			// TODO (SNOW-1737932): Setting schema parameters related to retention time seems to have no affect on streams on directory tables.
```

Comment:
let's find other way to make the stream stale then in this case

Reply:
I haven't found any. Do you have anything on your mind?

Reply:
I would experiment with changing the stage in some breaking ways. and additionally, I would just ask the stream team how we can make this particular type of stream stale.

Reply:
For now, I've added this behavior to our list and reported it in a streams channel. We test this logic in other resources, so I think there's little chance for a bug here. Also, we have a follow-up Jira on our side.